engine/src/cmd/collide_map.h

```diff
@@ -45,7 +43,7 @@ class Collidable {
         Unit *unit;
         unsigned int bolt_index;
     }
-            ref;
+            ref{};
```

Comment:
What does this do? I assume it's defined either way.
Also, can we fix the indentation. It's handing out there.

Reply:
Adding the empty braces initializes the value of `ref`. It was uninitialized previously, as far as I can tell.

Reply:
C++11 initialization method; we've generally stuck to constructor initialization but we've started getting more C++11 brace initialization in. We just have to make sure it's C++11 format and not C++20 feature in the process.

Reply:
Intuitively, it seemed to me it was initialized. However, on closer inspection, this is a non-static variable, so it really is uninitialized. it would be clearer if we were to specify :

`Unit *unit = nullptr;
        unsigned int bolt_index;
 } ref;`

Reply:
This initialization is for any kind of variable, not simply non-static. The C++ committee's idea is that the constructors don't need to do all the initialization of the class since they have this other means.
I think Strostroup's idea is that the constructor isn't necessary in 99% of cases and this lets them do that.

That said, I still kind of like having explicit constructors doing all the initialization.